Std/Lean/Name.lean

```diff
@@ -33,3 +36,44 @@ def isInternalDetail : Name → Bool
       || p.isInternalOrNum
   | .num _ _     => true
   | p            => p.isInternalOrNum
+
+end Lean.Name
+
+open Lean Meta Elab
+
+/-!
+# Additional functions on `Lean.Name`.
+-/
+
+private def isBlackListed (declName : Name) : CoreM Bool := do
+  if declName.toString.startsWith "Lean" then return true
+  let env ← getEnv
+  pure $ declName.isInternalDetail
+   || isAuxRecursor env declName
+   || isNoConfusion env declName
+  <||> isRec declName <||> isMatcher declName
```

Comment:
this definition is too tailored to the use case. I don't think it should go in std, and dependent functions should either be parameterized over it or they should also be removed from std.

Reply:
I removed it. can easily be added in the function `p : Name → Bool` (in mathlib) which is provided to the two commands.

Reply:
acutally, I moved all three declarations back into mathlib. Mainly because I realised it used `Std.HashMap` instead of `Lean.HashMap` which would complicate this PR further. Might upstream it in a future PR.